packages/trader/src/Stores/client-store.js

```diff
@@ -975,6 +980,29 @@ export default class ClientStore extends BaseStore {
         this.device_data = device_data;
     }
 
+    @action.bound
+    onSetResidence({ residence }, cb) {
+        if (!residence) return;
+        WS.setSettings({ residence: residence }).then(async response => {
+            if (response.error) {
+                cb(response.error.message);
+            } else {
+                await this.setResidence(residence);
+                await WS.authorized.storage.landingCompany(this.accounts[this.loginid].residence)
+                    .then(this.responseLandingCompany);
+                await WS.authorized.storage.getSettings().then(async response => {
+                    if (response.error) {
+                        cb(response.error.message);
```

Comment:
why callback the error here? the other previous async calls could fail too?

Reply:
To cover if it only fails when trying to call `getSettings`.